Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -86,17 +99,53 @@ public override string DataSource
             }
         }
 
-        public override string ServerVersion => _connectionState == ConnectionState.Open ? SfSession.serverVersion : "";
+        public override string ServerVersion => IsOpen() ? SfSession.serverVersion : "";
 
         public override ConnectionState State => _connectionState;
 
+        private object? GetTransactionId()

```

Comment:
are we able to better name the returned type?

Reply:
And this type should be?

Reply:
Maybe string, not sure of the CurrentTransaction return type, maybe some Int64 would suite???

Reply:
String, int, whatever :)

Reply:
Done